testing/testing-utils/src/chain_clients.rs

```diff
@@ -0,0 +1,507 @@
+        .curr_rewards_calculation_end_timestamp()
+        .await
+        .unwrap();
+
+    let submit_tx = rewards_coordinator
+        .submitRoot(root, curr_rewards_calculation_end_timestamp + 1)
+        .send()
+        .await
+        .unwrap();
+    let submit_status = submit_tx.get_receipt().await.unwrap().status();
+    assert!(submit_status);
+
+    (root, claim)
+}
+
+/// Creates an operator set from an http_endpoint and an avs address
+pub async fn create_operator_set(http_endpoint: &str, avs_address: Address) {
+    let allocation_manager_addr = get_allocation_manager_address(http_endpoint.to_string()).await;
+    let default_signer = get_signer(FIRST_PRIVATE_KEY, http_endpoint);
+    let allocation_manager =
+        AllocationManager::new(allocation_manager_addr, default_signer.clone());
+    let registry_coordinator_addr =
+        get_registry_coordinator_address(http_endpoint.to_string()).await;
+    let service_manager_address = get_service_manager_address(http_endpoint.to_string()).await;
+    let service_manager =
+        MockAvsServiceManager::new(service_manager_address, default_signer.clone());
+    service_manager
+        .setAppointee(
+            default_signer.default_signer_address(),
+            allocation_manager_addr,
+            alloy::primitives::FixedBytes(AllocationManager::setAVSRegistrarCall::SELECTOR),
+        )
+        .send()
+        .await
+        .unwrap()
+        .get_receipt()
+        .await
+        .unwrap();
+    allocation_manager
+        .setAVSRegistrar(avs_address, registry_coordinator_addr)
+        .send()
+        .await
+        .unwrap()
+        .get_receipt()
+        .await
+        .unwrap();
+
+    // Create slashable quorum
+    let contract_registry_coordinator =
+        RegistryCoordinator::new(registry_coordinator_addr, default_signer.clone());
+    let operator_set_params = OperatorSetParam {
+        maxOperatorCount: 10,
+        kickBIPsOfOperatorStake: 100,
+        kickBIPsOfTotalStake: 1000,
+    };
+    let strategy = get_erc20_mock_strategy(http_endpoint.to_string()).await;
+    service_manager
+        .setAppointee(
+            registry_coordinator_addr,
+            allocation_manager_addr,
+            alloy::primitives::FixedBytes(AllocationManager::createOperatorSetsCall::SELECTOR),
+        )
+        .send()
+        .await
+        .unwrap()
+        .get_receipt()
+        .await
+        .unwrap();
+    let strategy_params = StrategyParams {
+        strategy,
+        multiplier: U96::from(1),
+    };
+
+    contract_registry_coordinator
+        .createSlashableStakeQuorum(operator_set_params, U96::from(0), vec![strategy_params], 0)
+        .send()
+        .await
+        .unwrap()
+        .get_receipt()
+        .await
+        .unwrap();
+}
+
+/// Registers an operator to the received quorum numbers, with a private key to an anvil on the
+/// http_url using the avs_writer
+pub async fn test_register_operator(
```

Comment:
Let's rename this to just `register_operator_in_avs` (here or in a following PR).